packages/core/src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,101 @@
+import { Typography } from '@mui/material';
+import { Trans } from '@lingui/macro';
+import React, { Component, type ReactNode } from 'react';
+import StackTrace from "stacktrace-js";
+import { styled } from '@mui/styles';
+import qs from 'qs';
+import LayoutHero from '../LayoutHero';
+import Button from '../Button';
+import Link from '../Link';
+import Flex from '../Flex';
+
+const StyledPre = styled(Typography)(() => ({
+  whiteSpace: 'pre-wrap',
+}));
+
+function formatStackTrace(stack: []) {
+  const stackTrace = stack.map(({ fileName, columnNumber, lineNumber, functionName }) => {
+    return `at ${fileName}:${lineNumber}:${columnNumber} ${functionName}`;
+  });
+  return stackTrace.join('\n');
+}
+
+type ErrorBoundaryProps = {
+  children?: ReactNode;
+};
+
+export default class ErrorBoundary extends Component {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = {
+      hasError: false,
+      error: null,
+      stacktrace: '',
+    };
+  }
+
+  static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    return {
+      hasError: true,
+      error,
+    };
+  }
+
+  async componentDidUpdate(prevProps, prevState) {
+    if (this.state.error && prevState.error !== this.state.error) {
+      this.setState({
+        stacktrace: formatStackTrace(await StackTrace.fromError(this.state.error)),
+      });
+    }
+  }
+
+  handleReload = () => {
+    window.location.hash = '#/';
+    window.location.reload();
+  }
+
+  render() {
+    if (this.state.hasError) {
+      const { stacktrace, error } = this.state;
+      const issueLink = `https://github.com/Chia-Network/chia-blockchain/issues/new?${qs.stringify({
+        labels: 'bug,GUI',
+        template: 'bug_report.yaml',
+        title: `[BUG] ${error.message}`,
+        ui: 'GUI',
+        logs: `${error.message}\n\n${stacktrace}`,
```

Comment:
Would it be possible to include window.location or something to indicate where the exception originated (beyond what's in the stacktrace, which isn't always useful). Maybe a `useLocation()` hook would be useful?

Reply:
I added current URL.